app/controllers/groups_controller.rb

```diff
@@ -184,6 +184,58 @@ def csv_upload
     redirect_to action: 'index', id: params[:id]
   end
 
+  def create_groups_when_students_work_alone
+    @assignment = Assignment.find_by_id(params[:assignment_id])
+
+    if @assignment.group_max == 1
+      Student.all.map do |student|
+        # Check to see if the student already has a grouping for
+        # the current assignment
+        grouping = student.accepted_grouping_for(@assignment.id)
+        next unless grouping.nil?
+
+        ActiveRecord::Base.transaction do
+          grouping = Grouping.new
+          grouping.assignment_id = @assignment.id
+
+          # If an individual repo has already been created for this user
+          # then just use that one.
+          group = Group.find_by group_name: student.user_name
+          if group.nil?
+            group = Group.new(group_name: student.user_name)
+            group.repo_name = student.user_name
+            group.save
+            unless group.errors[:base].blank?
+              # raise an error and continue
+              collision_error = I18n.t('csv.repo_collision_warning',
+                                       repo_name: group.errors[:base],
+                                       group_name: group.group_name)
+              flash_message(:error,
+                            'Student ' + student.user_name + ': ' + \
+                             collision_error)
+            end
+          end
+
+          grouping.group = group
+          grouping.save
+          # Create the membership
+          member = StudentMembership.new(
+            grouping_id: grouping.id,
+            membership_status: StudentMembership::STATUSES[:inviter],
+            user_id: student.id)
+          member.save
+          # Update repo permissions if need be. This has to happen
+          # after memberships have been established.
+          grouping.update_repository_permissions
+          # Add permissions for TAs and admins.  This completely rerwrites the
+          # auth file but that shouldn't be a big deal in this case.
+          group.set_repo_permissions
```

Comment:
@david-yz-liu @reidka I took out these two repo permission statements and called Repository::SubversionRepository.__generate_authz_file.
I believe this method sets authorizations for all groups and I placed it after creating all the groups. 
The changes are in the pull request #2383